orc.js

```diff
@@ -1,42 +1,169 @@
-var app = require('commander');
+var commander = require('commander');
 var git = require('gitty');
 var path = require('path');
-
-// process.stdin.resume();
-// process.stdin.setEncoding('utf8');
-// process.stdin.on('data', function(data) {
-//  process.stdout.write(data);
-// });
+var pushover   = require('pushover');
 
 process.on('SIGINT', function () {
   console.log('Got a SIGINT. Goodbye cruel world');
   process.exit(0);
 });
 
-function _requireCommit() {
+function init(dir, cb) {
+  // TODO: march up to root dir just to make output consistent
+  var repo = git(dir);
+  repo.getBranches(function(err, branches) {
+    if(err) {
+      cb('error getting branches '+err);
+      return;
+    }
+    if (!branches.current) {
+      cb('error: '+repo.name+' doesn\'t appear to be a valid git repo');
+      return;
+    }
+    cb(null, { repo_dir: dir, repo: git(dir)});
+  });
+}
+
+function noUnstaged(orc, cb) {
+  orc.repo.status(function(err, status) {
+    if(err) {
+      cb('error getting status checking unstaged '+err);
+      return;
+    }
+    if(status.unstaged.length > 0) {
+      cb('unstaged files, commit before running this.');
+      return;
+    }
+    cb(null, true);
+  });
+}
+
+function noUntracked(orc, cb) {
+  orc.repo.status(function(err, status) {
+    if(err) {
+      cb('error getting status checking unstaged '+err);
+      return;
+    }
+    if(status.untracked.length > 0) {
+      cb('untracked files, add or ignore before running this.');
+      return;
+    }
```

Comment:
yet again, why not make this a helper function since you're using it in 2 places?